sktime/forecasting/model_selection/tests/test_tune.py

```diff
@@ -348,3 +351,87 @@ def test_gscv_backends(backend_set):
         backend_params=backend_params,
     )
     gscv.fit(y, X)
+
+
+def test_BaseGridSearch(
+    Forecaster,
+    *,
+    as_distribution=False,
+    **kwargs,
+):
+    if "param_distributions" not in kwargs and "param_grid" not in kwargs:
+        params = (
+            [
+                {
+                    "window_length": [1, 2, 3],
+                    "strategy": ["last", "mean"],
+                    "transformer__degree": [1, 2, 3],
+                    "forecaster__strategy": ["last", "mean", "seasonal_last"],
+                },
+                {
+                    "window_length": [4, 5, 6],
+                    "forecaster__strategy": ["last", "mean"],
+                },
+            ],
+        )
+        kwargs.setdefault(
+            "param_distributions" if as_distribution else "param_grid", params
+        )
+    return Forecaster, {
+        **kwargs,
+    }
+
+
+def test_ForecastingGridSearchCV(**kwargs):
+    return test_BaseGridSearch(ForecastingGridSearchCV, **kwargs)
+
+
+def test_ForecastingRandomizedSearchCV(**kwargs):
+    return test_BaseGridSearch(
+        ForecastingRandomizedSearchCV, as_distribution=True, **kwargs
+    )
+
+
+def test_ForecastingSkoptSearchCV(**kwargs):
+    return test_BaseGridSearch(ForecastingSkoptSearchCV, as_distribution=True, **kwargs)
+
+
+@pytest.mark.parametrize("return_n_best_forecasters", [-1, 0, 1, 2])
+@pytest.mark.parametrize(
+    "Forecaster, kwargs",
+    [
+        test_ForecastingGridSearchCV(),
+        test_ForecastingRandomizedSearchCV(),
+        test_ForecastingRandomizedSearchCV(n_iter=3),
```

Comment:
Just trying to understand, are these two cases different from precvious no specification one?

Reply:
As was suggested [here](https://github.com/sktime/sktime/pull/6031#discussion_r1506895681). I was trying to add tests for `ForecastingRandomizedSearchCV` so I think yes, they are testing different models even if they have the same base one.

Th first tests `ForecastingGridSearchCV`

Th second and third tests `ForecastingRandomizedSearchCV`
The second one runs with default `n_iter=10` and the third `n_iter=3`

Reply:
My question was between no specification and separately passing n_iter twice. In my understanding, it's enough to test any one of these 3 cases. There's no harm in adding more test cases,  but our CI takes a long time so that may help there slightly.